example-app/src/main/kotlin/org/eclipse/keyple/coppernic/ask/example/activity/MainActivity.kt

```diff
@@ -150,20 +142,10 @@ class MainActivity : AbstractExampleActivity(), PowerListener {
 //        samReader = samPlugin.getReader(AndroidFamocoReader.READER_NAME)
     }
 
-    override fun onResume() {
-        super.onResume()
-        addActionEvent("Enabling NFC Reader mode")
-        //poReader.enableNFCReaderMode(this)
-        addResultEvent("Please choose a use case")
-        if (areReadersInitialized.get()) {
-            (poReader as AndroidCoppernicAskContactlessReaderImpl).startCardDetection(ObservableReader.PollingMode.REPEATING)
-        }
-    }
-
     override fun onPause() {
         addActionEvent("Stopping PO Read Write Mode")
         if (areReadersInitialized.get()) {
-            (poReader as AndroidCoppernicAskContactlessReaderImpl).stopCardDetection()
+            (poReader as Cone2ContactlessReaderImpl).stopCardDetection()
```

Comment:
Maybe
   (poReader as ObservableReader).stopCardDetection()

Cone2ContactlessReaderImpl class should be defined package private